src/proto/tagpost_rpc.proto

```diff
@@ -44,3 +48,22 @@ message FetchCommentsUnderThreadRequest {
 message FetchCommentsUnderThreadResponse {
   repeated Comment comment = 1;
 }
+
+message AddThreadWithPrimaryTagRequest {
+  string tag = 1;
+}
+
+message AddThreadWithPrimaryTagResponse {
+  Thread thread = 1;
+}
+
+message AddCommentUnderThreadRequest {
```

Comment:
For this one, you can add a "Comment" object here, and remove duplicated fields, because this input is a little bit complex. Remember to add comments here to describe what fields are required and what fields should not be filled.